seqr/views/apis/family_api.py

```diff
@@ -269,15 +269,20 @@ def update_family_fields_handler(request, family_guid):
     check_project_permissions(family.project, request.user)
 
     request_json = json.loads(request.body)
+    immutable_keys = [] if external_anvil_project_can_edit(family.project, request.user) else ['family_id']
     update_family_from_json(family, request_json, user=request.user, allow_unknown_keys=True, immutable_keys=[
-        'family_id', 'display_name',
-    ])
+        'display_name',
+    ] + immutable_keys)
 
     return create_json_response({
-        family.guid: _get_json_for_model(family, user=request.user)
+        family.guid: _get_json_for_model(family, user=request.user, process_result=_set_display_name)
     })
 
 
+def _set_display_name(family_json, family_model):
```

Comment:
In `update_family_from_json` there's also a few lines that set `displayName` in the json response. Do you need both?

Reply:
yeah, so in that its converting the json to be used to update the model in the database and set the fields correctly, here its pulling the data from the database and formatting it into json. So its the reverse